src/CalculatorUITestFramework/WindowsElementExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+using OpenQA.Selenium.Appium.Windows;
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace CalculatorUITestFramework
+{
+    public static class WindowsElementExtensions
+    {
+        /// <summary>
+        /// Waits for an element to be displayed until the timeout is reached.
+        /// </summary>
+        /// <param name="element">WindowsElement in the Calculator application.</param>
+        /// <param name="timeout">Timeout in ms.</param>
+        public static void WaitForDisplayed(this WindowsElement element, int timeout = 2000)
+        {
+            Stopwatch timer = new Stopwatch();
+            timer.Reset();
```

Comment:
Is this line needed?

Reply:
The timer was not being reset properly without explicitly calling it.

Reply:
How do you know? This should not be needed--I'm wondering if there's something else happening.

---
In reply to: [294546554](https://github.com/microsoft/calculator/pull/501#discussion_r294546554) [](ancestors = 294546554)